cv_next/app/auth/callback/route.ts

```diff
@@ -0,0 +1,37 @@
+import { cookies } from "next/headers"
+import { NextResponse } from "next/server"
+import { type CookieOptions, createServerClient } from "@supabase/ssr"
+
+export async function GET(request: Request) {
+  const { searchParams, origin } = new URL(request.url)
+  const code = searchParams.get("code")
+  // if "next" is in param, use it as the redirect URL
+  const next = searchParams.get("next") ?? "/"
+  if (code) {
+    const cookieStore = cookies()
+    const supabase = createServerClient(
+      process.env.NEXT_PUBLIC_SUPABASE_URL!,
+      process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY!,
+      {
+        cookies: {
+          get(name: string) {
+            return cookieStore.get(name)?.value
+          },
+          set(name: string, value: string, options: CookieOptions) {
+            cookieStore.set({ name, value, ...options })
+          },
+          remove(name: string, options: CookieOptions) {
+            cookieStore.delete({ name, ...options })
+          },
+        },
+      }
+    )
+    const { error } = await supabase.auth.exchangeCodeForSession(code)
+    if (!error) {
+      return NextResponse.redirect(`${origin}${next}feed`)
+    }
+  }
+
+  // return the user to an error page with instructions
+  return NextResponse.redirect(`${origin}/auth/auth-code-error`)
```

Comment:
Does this route exist?

Reply:
We don't, it redirect's to next's not found
What instructions should this page even contain? It only get's here if the code param wasn't found

Reply:
Well, I guess it's an example code right? The point is, don't leave these kind of lines if they don't match any of our actual use case. If we want, we can create this route and handle this case. But if we don't want to handle such case in an informative way, we can simply redirect to error page instead of populating an inexistent route to our codebase.

Reply:
This only happens when Google can't authenticate the user(no oAuth code was returned)...
Is that something worth handling?

Reply:
Well sounds like a possible timeout error, etc. Either redirect to some error page we have, or throw to homepage for now.
If we *know* there's a possibility a user will encounter some flow, we'd better handle it. The only possibilities we leave out are those we aren't aware of :)
(this could also be dealt with in a toast-like behavior on the auth page, but it'll require a bit more changes than we actually want right now)

Reply:
Redirecting to the home page, added a todo to redirect to an error page once it is merged